### PR TITLE
Fix conda build

### LINKED ISSUE #20172: conda build error
### Expected behavior
successful conda linux-64 build for mantid-framework

### Actual behavior
Failing due to missing definition of DBL_MAX

See http://builds.mantidproject.org/job/master_create_conda_linux_pkg/188/console or http://builds.mantidproject.org/job/master_create_conda_linux_pkg/212/console

```
03:58:44 /build_artefacts/mantid-framework_1501818793778/work/Framework/Geometry/src/Surfaces/Cylinder.cpp: In member function ‘virtual void Mantid::Geometry::Cylinder::getBoundingBox(double&, double&, double&, double&, double&, double&)’:
03:58:44 /build_artefacts/mantid-framework_1501818793778/work/Framework/Geometry/src/Surfaces/Cylinder.cpp:400:26: error: ‘DBL_MAX’ was not declared in this scope
03:58:44      xmin = ymin = zmin = DBL_MAX;
03:58:44                           ^
```
### Steps to reproduce the behavior
Run docker

    $ docker run -it condaforge/linux-anvil

In docker,

    $ git clone https://github.com/mantidproject/conda-recipes
    $ cd conda-recipes/frameworks

Modify meta.yaml, so that 

    version = "3.10.debug"
    git_rev = "f023278835accb47aaf2e2b58c959365dc525579"

Build:

    export OPENGL_gl_LIBRARY=/usr/lib64/libGL.so 
    export OPENGL_glu_LIBRARY=/usr/lib64/libGLU.so 
    yum install -y mesa-libGLU-devel 
    conda build --python 2.7 --numpy 1.13 &>log.build &
    tail -f log.build

### Platforms affected
conda

### PR DESCRIPTION
**Pull request by yxqd**
Description of work.

**To test:**

Run docker

    $ docker run -it condaforge/linux-anvil

In docker,

    $ git clone https://github.com/mantidproject/conda-recipes
    $ cd conda-recipes/frameworks

Modify meta.yaml, so that 

    version = "3.10.debug"
    git_rev = "fix-conda-build"

Build:

    export OPENGL_gl_LIBRARY=/usr/lib64/libGL.so 
    export OPENGL_glu_LIBRARY=/usr/lib64/libGLU.so 
    yum install -y mesa-libGLU-devel 
    conda build --python 2.7 --numpy 1.13 &>log.build &
    tail -f log.build

Fixes #20172 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Kernel/inc/MantidKernel/V3D.h [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #define MANTID_KERNEL_V3D_H_
 
 #include <cmath>
+#include <cfloat>
```
Comment (quantumsteve) [inline Framework/Kernel/inc/MantidKernel/V3D.h]: Why are you adding this to `V3D.h` when the undefined macro is in `Cylinder.cpp`?

**Review by quantumsteve — CHANGES_REQUESTED** [review]
Please move the new include to Cylinder.cpp

**Comment by peterfpeterson** [comment]
You should change to [`std::numeric_limits<double>::max()`](https://stackoverflow.com/questions/23278930/what-is-dbl-max-in-c#23279161) since we're a C++ project.

File: Framework/Kernel/inc/MantidKernel/V3D.h (hunk 1 above)
Comment (yxqd) [inline Framework/Kernel/inc/MantidKernel/V3D.h]: Rolled back

**Comment by yxqd** [comment]
@quantumsteve @peterfpeterson please check

**Review by peterfpeterson — APPROVED** [review]
Looks good to me.

**Review by quantumsteve — APPROVED** [review]
:+1:

**Pull request merged**